kickstarter/android-oss
# MBL-1768: RegisterTokenWorker to RXJava2

# 📲 What

- One more migration to RXJava2

# 🛠 How
- Reusable code by bringin `RefreshPushToken` interface implementation used as well in `MessageService` 

# 👀 See
- Take a look at [datadog](https://app.datadoghq.com/logs?query=%40http.useragent%3A%2AKickstarter%2AAndroid%2A%20%40environment%3Astaging%20%40controller%3A%22Api%3A%3AV1%3A%3APushTokensController%22%20&agg_m=count&agg_m_source=base&agg_t=count&cols=host%2Cservice%2C%40http.url_details.queryString.oauth_token&fromUser=true&messageDisplay=inline&refresh_mode=sliding&storage=hot&stream_sort=desc&viz=stream&from_ts=1618865525266&to_ts=1621457525266&live=true)
- Log in fresh in the app, you should see calls to `push_tokens` endpoint

<img width="1279" alt="Screenshot 2024-10-09 at 4 39 06 PM" src="https://github.com/user-attachments/assets/8264b207-2acc-4c02-bc3a-0ab304fe323e">

| --- | --- |
|  |  |

# 📋 QA
- Make sure you are logged out
- Take a look at datadog or you Android Studio network profiler
- See the calls to `push_token` endpoint

# Story 📖

[MBL-1768](https://kickstarter.atlassian.net/browse/MBL-1768)


[MBL-1768]: https://kickstarter.atlassian.net/browse/MBL-1768?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
